### PR TITLE
Remove qtviewer from viewer

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
This PR would address more of our MVC refactor for example #92 by removing the `_qtviewer` from the `ViewerWidget` which will be renamed `ViewerModel`. It is currently not running as I need to fix some imports

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# How has this been tested?

- [x] run examples

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by sofroniewn** [comment]
Bugs have now been fixed and this is ready for review. One thing I'll note is that `ViewerModel` still has a property on it `_scene` that gets set after it is passed to `QtViewer` which is needed for the individual `Layer` model files to access the scenecanvas. Once these have been refactored too this can be removed

File: napari/components/_viewer/model.py [hunk 1]
```diff
@@ -59,30 +60,15 @@ def __init__(self, title='napari'):
         self.key_bindings = {}
 
         # TODO: this should be eventually removed!
-        # initialised by QtViewer when it is constructed by the model
-        self._qtviewer = None
+        # attaced by QtViewer when it is constructed by the model
```
Comment (kne42) [inline napari/components/_viewer/model.py]:
```suggestion
        # attached by QtViewer when it is constructed by the model
```
also is it necessary to have this on here? if it's `None`, none of the related code should run anyways

File: napari/components/_viewer/view/main.py [hunk 2]
```diff
@@ -73,7 +76,42 @@ def __init__(self, viewer):
                 'standard': QCursor()
             }
 
-    def set_cursor(self, cursor, size=10):
+        self.viewer.events.interactive.connect(self._on_interactive)
+        self.viewer.events.cursor.connect(self._on_cursor)
+        self.viewer.events.reset_view.connect(self._on_reset_view)
+        self.viewer.layers.events.reordered.connect(self._update_canvas)
+
+    def screenshot(self, region=None, size=None, bgcolor=None):
```
Comment (kne42) [inline napari/components/_viewer/view/main.py:84]: What's the point of moving this here if no one is going to be able to use it?

Comment (kne42) [reply napari/components/_viewer/view/main.py:84]: I think this goes back to what @royerloic was hinting at during yesterday's meeting, that we need a controller class that combines the model and the given view (depending on backend) while also providing convenience functionality for the user.

**Review by kne42 — APPROVED** [review]
LGTM

File: napari/components/_viewer/model.py (hunk 1 above)
Comment (sofroniewn) [reply napari/components/_viewer/model.py]: typo fixed. this is only initialized to `None`, it gets set to be the `scene` inside the `QtViewer`, but we will remove this in the future

File: napari/components/_viewer/view/main.py (hunk 2 above)
Comment (sofroniewn) [reply napari/components/_viewer/view/main.py:84]: for now I forwarded the `screenshot` method inside the `Viewer` object so it's still accessible. Agreed a controller might be needed. Also now it's a little strange that `to_svg` is still on `ViewerModel`, but `screenshot` is on `QtViewer`. Maybe we can fix in future PRs though

File: napari/components/_viewer/model.py (hunk 1 above)
Comment (kne42) [reply napari/components/_viewer/model.py]: Yes, but why is it initialized at all? It's going to break regardless; I don't think it belongs here.

Comment (sofroniewn) [reply napari/components/_viewer/model.py]: What do you mean by "break"? It runs fine now. The reason I put it here was that it gets used again below, line 248 `layer._parent = self._scene`. I'm not sure where else that could be

File: napari/components/_viewer/view/main.py (hunk 2 above)
Comment (sofroniewn) [reply napari/components/_viewer/view/main.py:84]: `screenshot` is also able to be called from `Viewer`, there are no issues with `self`

File: napari/components/_viewer/model.py (hunk 1 above)
Comment (kne42) [reply napari/components/_viewer/model.py]: ah, ok I didn't notice that. this was more a callback to the initial dims refactor PR which introduced this (and wasn't necessary at the time)

File: napari/components/_viewer/view/main.py (hunk 2 above)
Comment (kne42) [reply napari/components/_viewer/view/main.py:84]: yes, when you first told me this i thought you had forwarded it on the class level but the instance level works fine

**Pull request merged**